code/workhorse/OOPGP.R

```diff
@@ -0,0 +1,118 @@
+# Define a class named Simulator.
+# The main method used in main.R is the simulate method. Other methods of this class and modules in other files are helper functions 
+# for the simulate method.
+
+
+library(mlegp)
+Simulator <- function(file_path_,setting_){
+  setting = setting_
+  input_path = file_path_
+  dists = NA 
+  gene_id = 15 
+  res_path = NA
+  res_name = NA
+  res_path = NA
+  data_all = NA
+  train = NA
+  train_backup = NA
+  pool = NA
+  pool_all  = NA
+  benchmark = NA
+  model = NA
+  errors = NA
+  structure(class="Simulator", environment())
+}
+
+
+source("prepare_data.R")
+
+train_simulator <- function(simulator) {
+    train = simulator$train[, c(1:14, simulator$gene_id)]
+    print("training data gene id")
+    print(dim(train))
+    num = ncol(train)
+    simulator$model = mlegp(train[,2:num-1], train[,num])
+    
+  }
+  
+
+source("update_train_pool.R")
+
+simulate_index <- function(simulator, metric_fun){
+   select_index = as.numeric()
+   for (i in 1:simulator$setting[["add"]]) {
+        metric = metric_fun(simulator)
+        index=sort(metric, decreasing = TRUE,index.return=TRUE)$ix
+   
+        select_index = c(select_index,index[1])
+        update_train_by_prediction(simulator, index[1])
+        train_simulator(simulator)
+   }
+   return (select_index)
+}
+
+
+source("screen_index_helper.R")
+source("max_dist.R")
+
+screen_index <- function(simulator) {
+  method = simulator$setting[["method"]]
+  if (method != "Random" && method != "Expert") {
+      metric_fun = switch(simulator$setting[["method"]],
```

Comment:
It seems better to define a `get_metric_fun` function instead.

Reply:
This module is only used once. It would be necessary to put it in a function if it were used in multiple places.

Reply:
Correct, that's one of the reasons for adding a function, the other one being readability.